Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -387,9 +394,158 @@ namespace Tensile
+                                     m_powerValues.end());
+                    return static_cast<double>(m_powerValues[midValueIndex]);
+                }
+                break;
+            case GPU_TEMPERATURE_AVG:
+                if(m_tempHotspotValues.size() != 0)
+                {
+                    return (std::accumulate(
+                               m_tempHotspotValues.begin(), m_tempHotspotValues.end(), double(0)))
+                           / m_tempHotspotValues.size();
+                }
+                break;
+            case GPU_TEMPERATURE_MEDIAN:
+                if(m_tempHotspotValues.size() != 0)
+                {
+                    size_t midValueIndex = m_tempHotspotValues.size() / 2;
+                    std::nth_element(m_tempHotspotValues.begin(),
+                                     m_tempHotspotValues.begin() + midValueIndex,
+                                     m_tempHotspotValues.end());
+                    return static_cast<double>(m_tempHotspotValues[midValueIndex]);
+                }
+                break;
+            default:
+                throw std::runtime_error("Invalid statistical type index");
+                return std::numeric_limits<double>::quiet_NaN();
+            }
+            throw std::runtime_error(concatenate("Can\'t read  ",
+                                                 m_freqPowerTempErrorsStrings[freqPowerTempIndex],
+                                                 " value, empty or value wasn't requested: "));
+        }
+
+        void HardwareMonitor::logMinMaxMedianAverage()
+        {
+            assertNotActive();
+            if(m_hasInvalidGpuMetricStatus)
+                throw std::runtime_error("Invalid GPU metric data!");
+            if(m_dataPoints == 0)
+                throw std::runtime_error("No data points collected!");
+
+            if(m_freqValues.size() == 0 || m_powerValues.size() == 0
+               || m_tempHotspotValues.size() == 0)
+            {
+                throw std::runtime_error("Frequency, Power, Temperature array is empty.\n");
+            }
+
+            std::cout << "\nROCm SMI API consolidated frequency,power,temperature data"
+                      << "\n";
+            std::cout << "GFX Value\t\t\t\t"
+                      << "PPT0_value\t\t\t"
+                      << "Temperature\n";
+            // Log individual data
+            for(int i = 0; i < m_freqValues.size(); i++)
+            {
+                std::cout << std::left << std::setw(13) << m_freqValues[i] << "\t\t\t\t"
+                          << m_powerValues[i] << "\t\t\t\t\t" << m_tempHotspotValues[i] << "\n";
+            }
+
+            std::cout << "\n";
+            std::cout << "\t\t\t\t\tMin\t\t"
+                      << "Max\t\t"
+                      << "Average\t\t"
+                      << "Median\n";
+
+            // min, max,avg, median frequency
+            std::cout << std::left << std::setw(20) << "GFX Frequency" << std::setw(8)
```

Comment:
Also, unify this part as well.
We can create a sub function and pass a reference of vector.